ext/node/ops/buffer.rs

```diff
@@ -119,3 +120,63 @@ fn utf8_to_ascii(source: &[u8]) -> Vec<u8> {
   }
   ascii_bytes
 }
+
+#[op2]
+pub fn op_node_decode_utf8<'a>(
+  scope: &mut v8::HandleScope<'a>,
+  buf: v8::Local<v8::Value>,
+  byte_offset: v8::Local<v8::Value>,
+  byte_length: v8::Local<v8::Value>,
+  start: v8::Local<v8::Value>,
+  end: v8::Local<v8::Value>,
+) -> Result<v8::Local<'a, v8::String>, JsErrorBox> {
+  // SAFETY: the javascript side must guarantee that the arguments are of the correct types
+  let buf = unsafe { v8::Local::<v8::ArrayBuffer>::cast_unchecked(buf) };
+  // SAFETY: upheld by js
+  let byte_offset = unsafe { to_usize_unchecked(byte_offset) };
+  // SAFETY: upheld by js
+  let byte_length = unsafe { to_usize_unchecked(byte_length) };
+  // SAFETY: upheld by js
+  let start = unsafe { to_usize_unchecked(start) };
+  // SAFETY: upheld by js
+  let end = unsafe { to_usize_unchecked(end) };
```

Comment:
We don't generally let JS uphold Rust side safety rules, because we assume that anyone can call any op on the JS side.

How necessary is it to not do any bounds checking here?

Reply:
If we think that we have isolated away ops well enough to make this tradeoff, at minimum ops that require their contract to be upheld in JS should have `unsafe` in the name. And we should have a linter that requires a safety comment on JS uses of that op.

But really doing an unchecked byte read like this is just asking for trouble. This is like the second most powerful gadget we could provide to a bad actor after an arbitrary jump.

How is it possible that the same checks take more time when run in Rust instead of JS?

Reply:
Discussed and found a path forwards that allows us to remove the unsafety while keeping the same perf